english/code-generation.md

```diff
@@ -0,0 +1,52 @@
+Creation of a new module is one of the bottlenecks in VIPER, especially from the viewpoint of an outsider. The bare minimum needed in order to create a new module-screen is:
+
+- **Five new classes** (Assembly, ViewController, Presenter, Interactor, Router),
+- **Five new protocols** (ViewInput, ViewOutput, InteractorInput, InteractorOutput, RouterInput),
+- **Five new tests** (AssemblyTests, ViewControllerTests, PresenterTests, InteractorTests, RouterTests).
+
+Moreover, all the necessary relations need to be established, protocols need to be implemented, dependency injection set up - and plenty of other things in particular cases. This complexity brings two main problems:
+
+- Too much time spent on simple routine work
+- Increased probability of typos, which may harm not only code style but the application logic as well
+
+One of the ways to tackle this issue, which we also have been using in Rambler&Co - is making our own templates for Xcode. Such approach solves all aforementioned issues, but is has a number of its own disadvantages:
+
+* Building a new template is a complex process due to cumbersome syntax
+* Sometimes after an Xcode update templates might just stop working
+* No convenient way of adding new templates to Xcode (Alcatraz does not count)
+* No way to add template files to different targets (for example, when generating auto tests)
```

Comment:
Impossible to add template files to different targets (e.g. when generating auto tests)